jerry-core/jerry-api.h

```diff
@@ -146,6 +146,149 @@ typedef void (*jerry_object_free_callback_t) (const uintptr_t native_p);
+ */
+extern EXTERN_C bool
+jerry_api_value_is_undefined (const jerry_api_value_t *value_p);
+
+/**
+ * Returns whether the given jerry_api_value_t has boolean type.
+ */
+extern EXTERN_C bool
+jerry_api_value_is_boolean (const jerry_api_value_t *value_p);
+
+/**
+ * Returns whether the given jerry_api_value_t is number.
+ *
+ * More specifically, returns true if the type is JERRY_API_DATA_TYPE_FLOAT32,
+ * JERRY_API_DATA_TYPE_FLOAT64 or JERRY_API_DATA_TYPE_UINT32, false otherwise.
+ */
+extern EXTERN_C bool
+jerry_api_value_is_number (const jerry_api_value_t *value_p);
+
+/**
+ * Returns whether the given jerry_api_value_t is string.
+ */
+extern EXTERN_C bool
+jerry_api_value_is_string (const jerry_api_value_t *value_p);
+
+/**
+ * Returns whether the given jerry_api_value_t is object.
+ */
+extern EXTERN_C bool
+jerry_api_value_is_object (const jerry_api_value_t *value_p);
+
+/**
+ * Returns whether the given jerry_api_value_t is a function object.
+ *
+ * More specifically, returns true if the jerry_api_value_t of the value
+ * pointed by value_p has JERRY_API_DATA_TYPE_OBJECT type and
+ * jerry_api_is_function() functiron return true for its v_object member,
+ * otherwise false.
+ */
+extern EXTERN_C bool
+jerry_api_value_is_function (const jerry_api_value_t *value_p);
+
+/**
+ * Returns the boolean v_bool member of the given jerry_api_value_t structure.
+ * If the given jerry_api_value_t structure has type other than
+ * JERRY_API_DATA_TYPE_BOOLEAN, JERRY_ASSERT fails.
+ */
+extern EXTERN_C bool
+jerry_api_get_boolean_value (const jerry_api_value_t *value_p);
+
+/**
+ * Returns the number value of the given jerry_api_value_t structure
+ * as a double.
+ *
+ * If the given jerry_api_value_t structure has type JERRY_API_DATA_TYPE_UINT32
+ * v_uint32 member will be returned as a double value. If the given
+ * jerry_api_value_t structure has type JERRY_API_DATA_TYPE_FLOAT32
+ * v_float32 member will be returned as a double and if tpye is
+ * JERRY_API_DATA_TYPE_FLOAT64 the function returns the v_float64 member.
+ * As long as the type is none of the above, JERRY_ASSERT falis.
+ */
+extern EXTERN_C double
+jerry_api_get_number_value (const jerry_api_value_t *value_p);
+
+/**
+ * Returns the v_string member of the given jerry_api_value_t structure.
+ * If the given jerry_api_value_t structure has type other than
+ * JERRY_API_DATA_TYPE_STRING, JERRY_ASSERT fails.
+ */
+extern EXTERN_C jerry_api_string_t *
+jerry_api_get_string_value (const jerry_api_value_t *value_p);
+
+/**
+ * Returns the v_object member of the given jerry_api_value_t structure.
+ * If the given jerry_api_value_t structure has type other than
+ * JERRY_API_DATA_TYPE_OBJECT, JERRY_ASSERT fails.
+ */
+extern EXTERN_C jerry_api_object_t *
+jerry_api_get_object_value (const jerry_api_value_t *value_p);
+
+/**
+ * Creates and returns a jerry_api_value_t with type
+ * JERRY_API_DATA_TYPE_VOID.
+ */
+extern EXTERN_C jerry_api_value_t
+jerry_api_create_void_value ();
```

Comment:
AFAIK: our stlye "dictates" that we when there is no argument for a function we should add the `void`, for example this should be:
```
jerry_api_create_void_value (void);
```